### PR TITLE
ICU-21480 BRS-50: updated docmain.h

### PR DESCRIPTION
**Pull request by sven-oly**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21480
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by sven-oly** [comment]
This is my first attempt at updating this file based on API Updates. Please take a look and comment on these for correctness and completeness. Tnx.

File: icu4c/source/common/unicode/docmain.h [hunk 1]
```diff
@@ -138,6 +138,11 @@
  *     <td>icu::DateFormat</td>
  *   </tr>
  *   <tr>
+ *     <td>Basic Time Zone, Rule Based Time Zone, Simple Time Zone, VTimeZone</td>
+ *     <td>basictz.h, rbtz.h simpletz.h, vtzone.h</td>
+ *     <td>icu::BasicTimeZone, icu::RuleBasedTimeZone, icu::SimpeTimeZone, icu::VTimeZone</td>
```
Comment (markusicu) [inline icu4c/source/common/unicode/docmain.h:143]: We don't need to list the subclasses, but I am surprised that there is not already an entry for icu::TimeZone. I would either add that to "Calendars" (and rename to "Calendars & time zones") or make a row for just "time zones". There is no dedicated C API though; you can just repeat ucal.h for that if it's a separate row.

File: icu4c/source/common/unicode/docmain.h [hunk 2]
```diff
@@ -158,6 +163,11 @@
  *     <td>icu::number::NumberRangeFormatter</td>
  *   </tr>
  *   <tr>
+ *     <td>Measure Unit Formatting</td>
+ *     <td>measunit.h</td>
+ *     <td>icu::MeasureUnit</td>
```
Comment (markusicu) [inline icu4c/source/common/unicode/docmain.h:168]: These are just the units, not the formatting API. We already have "Number Formatting (includes currency and unit formatting)"

So I don't think we need this.

File: icu4c/source/common/unicode/docmain.h (hunk 1 above)
Comment (sven-oly) [reply icu4c/source/common/unicode/docmain.h]: Added to Calendars line.

File: icu4c/source/common/unicode/docmain.h (hunk 2 above)
Comment (sven-oly) [reply icu4c/source/common/unicode/docmain.h]: Removed MeasureUnit

File: icu4c/source/common/unicode/docmain.h [hunk 3]
```diff
@@ -74,7 +74,7 @@
  *   </tr>
  *   <tr>
  *     <td>Strings and Character Iteration</td>
- *     <td>ustring.h, utf8.h, utf16.h, UText, UCharIterator</td>
+ *     <td>ustring.h, utf8.h, utf16.h, bytestream.h, UText, UCharIterator</td>
```
Comment (markusicu) [inline icu4c/source/common/unicode/docmain.h:77]:
```suggestion
 *     <td>ustring.h, utf8.h, utf16.h, icu::StringPiece, UText, UCharIterator, icu::ByteSink</td>
```

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Pull request merged**